hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -1463,7 +1463,7 @@ public ICompletableFuture submitToKeyInternal(Object key, EntryProcessor entryPr
                                                    IMap map) {
         QueryCacheRequest request = newQueryCacheRequest()
                 .withUserGivenCacheName(name)
-                .withCacheName(UuidUtil.newUnsecureUuidString())
+                .withCacheUuid(UuidUtil.newUnsecureUuidString())
```

Comment:
withCacheId for the technical identifier 
withName for the 'user provided' name.

Simple naming is key to have simple software.